src/main/java/scala_maven/ScalaDocMojo.java

```diff
@@ -171,7 +179,13 @@ protected JavaMainCaller getScalaCommand() throws Exception {
                 .getOutputDirectory())); // remove output to avoid "error for" : error: XXX is
     // already defined as package XXX ... object XXX {
     addAdditionalDependencies(paths);
-    if (!paths.isEmpty()) jcmd.addOption("-classpath", FileUtils.toMultiPath(paths));
+    if (sc.version().major == 3) {
+      addScalaDocToClasspath(paths);
+    }
```

Comment:
The `if` test is useless, as `addScalaDocToClasspath` will add nothing if for empty list as you define into Context for Scala 2

Reply:
You are correct. I have removed the needless test.